### ISSUE TITLE
[BUG] After upgrade to v1.4.0-rc4, one image display 0% progress and can't be used anymore 

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#6894]
**Describe the bug**

After upgrade from v1.3.2 to v1.4.0-rc4, one of the image `sles15-sp4-minimal-vm.x86_64-openstack-cloud-gm.qcow2` 
display 0% progress with message

  ```
  Retry attempted 1/3 failed due to error: failed to process sync file: syncing file should be directly reused but failed
  ```
  ![image](https://github.com/user-attachments/assets/f3e96e61-7fc0-4d4e-8707-01a2b3961675)

* We can't find them in the image list 

  ![image](https://github.com/user-attachments/assets/6c15b696-599b-42ca-9162-a83fc2a08674)

**To Reproduce**
1. Prepare 4 nodes v1.3.2 Harvester cluster 
2. Create image `sles15-sp6-minimal-vm.x86_64-cloud-gm.qcow` and `sles15-sp4-minimal-vm.x86_64-openstack-cloud-gm.qcow2` from download URL
3. Create two virtual machines
4. Integrate with Rancher v2.7.15
5. Provision RKE2 downstream cluster
6. Upgrade Harvester to v1.4.0-rc4
7. Check the image state 

**Expected behavior**

After upgrade to v1.4.0-rc4, all images should keep the original state and can't become unusable. 

**Support bundle**

[supportbundle_93e674a6-935b-4b5b-a86e-815dbed50ba4_2024-10-27T12-23-07Z.zip](https://github.com/user-attachments/files/17533914/supportbundle_93e674a6-935b-4b5b-a86e-815dbed50ba4_2024-10-27T12-23-07Z.zip)

**Environment**
 - Harvester ISO version: v1.3.2 -> v1.4.0-rc4
 - Underlying Infrastructure: 4 nodes remote kvm machines

**Comment by WebberHuang1118** [comment]
By original design, vmi controller doesn't reconcile bi state once bi is created successfully.

However, as introducing virtual size to vmi, after the upgrade, vmi controller would reconcile the bi state, because virtual size is not updated yet https://github.com/harvester/harvester/blob/19bc22db86ec0c9d796ec1d9116c3ddce908082a/pkg/controller/master/image/backing_image_controller.go#L59-L61.

In this case, the above situation and bi `DiskFileStatusMap` restructure happen at the same time, and lead to the message `retry attempted 1/3 failed due to error: failed to process sync file: syncing file should be directly reused but failed`.

Not sure if the bi `DiskFileStatusMap` restructure is expected or not, since lack of logs, but we can make sure the bi disk file keeps sync from a valid one, for example:
  ```
  2024-10-27T06:51:05.453041134Z time="2024-10-27T06:51:05Z" level=info msg="Requesting syncing backing image" func="controller.(*BackingImageManagerController).prepareBackingImageFiles" file="backing_image_manager_controller.go:786" backingImage=vmi-aad51d2a-c8f3-4907-9fd0-64d77b29cbc7 backingImageManager=backing-image-manager-5ae6-4825 controller=longhorn-backing-image-manager diskUUID=48254039-3639-4916-8f77-068b9e663872 fromHost=10.52.3.37 node=harvester-node-2 nodeID=harvester-node-2 size=288577024
  
  2024-10-27T06:51:16.742957153Z time="2024-10-27T06:51:16Z" level=info msg="Deleting the file for invalid backing image vmi-aad51d2a-c8f3-4907-9fd0-64d77b29cbc7, in backing image manager spec UUID 0b3cdb34, backing image correct UUID 0b3cdb34" func="controller.(*BackingImageManagerController).deleteInvalidBackingImages" file="backing_image_manager_controller.go:642" backingImageManager=backing-image-manager-5ae6-4825 controller=longhorn-backing-image-manager diskUUID=48254039-3639-4916-8f77-068b9e663872 node=harvester-node-2 nodeID=harvester-node-2
  
  2024-10-27T06:51:17.783653328Z time="2024-10-27T06:51:17Z" level=info msg="Requesting syncing backing image" func="controller.(*BackingImageManagerController).prepareBackingImageFiles" file="backing_image_manager_controller.go:786" backingImage=vmi-aad51d2a-c8f3-4907-9fd0-64d77b29cbc7 backingImageManager=backing-image-manager-5ae6-4825 controller=longhorn-backing-image-manager diskUUID=48254039-3639-4916-8f77-068b9e663872 fromHost=10.52.3.37 node=harvester-node-2 nodeID=harvester-node-2 size=288577024
  
  ......
  
  2024-10-27T07:48:25.596851222Z time="2024-10-27T07:48:25Z" level=info msg="Requesting syncing backing image" func="controller.(*BackingImageManagerController).prepareBackingImageFiles" file="backing_image_manager_controller.go:786" backingImage=vmi-aad51d2a-c8f3-4907-9fd0-64d77b29cbc7 backingImageManager=backing-image-manager-5ae6-4825 controller=longhorn-backing-image-manager diskUUID=48254039-3639-4916-8f77-068b9e663872 fromHost=10.52.1.78 node=harvester-node-2 nodeID=harvester-node-2 size=288577024
  ```

and the bi is in a complete state from the SB
  ```
  apiVersion: longhorn.io/v1beta2
  kind: BackingImage
  metadata:
    annotations:
      harvesterhci.io/imageId: default/image-ql75b
      sim.harvesterhci.io/creationTimestamp: "2024-10-26T13:49:40Z"
    creationTimestamp: "2024-10-26T13:49:40Z"
    finalizers:
    - longhorn.io
    generation: 1
    labels:
      longhorn.io/component: backing-image
      longhorn.io/managed-by: longhorn-manager
    managedFields:
    - apiVersion: longhorn.io/v1beta2
      fieldsType: FieldsV1
      fieldsV1:
        f:spec:
          f:checksum: {}
      manager: harvester
      operation: Update
      time: "2024-10-26T13:49:40Z"
    - apiVersion: longhorn.io/v1beta2
      fieldsType: FieldsV1
      fieldsV1:
        f:spec:
          f:secret: {}
          f:secretNamespace: {}
      manager: longhorn-manager
      operation: Update
      time: "2024-10-27T09:07:53Z"
    - apiVersion: longhorn.io/v1beta2
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          f:diskFileStatusMap:
            f:0e762d26-0c69-47f2-9c65-8ab5e1723f44:
              f:message: {}
            f:48254039-3639-4916-8f77-068b9e663872:
              f:message: {}
            f:c8e8efeb-6c05-4854-b230-015ad8888ecc:
              f:message: {}
      manager: longhorn-manager
      operation: Update
      subresource: status
      time: "2024-10-27T09:08:12Z"
    - apiVersion: longhorn.io/v1beta2
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            .: {}
            f:harvesterhci.io/imageId: {}
            f:sim.harvesterhci.io/creationTimestamp: {}
          f:finalizers:
            .: {}
            v:"longhorn.io": {}
          f:labels:
            .: {}
            f:longhorn.io/component: {}
            f:longhorn.io/managed-by: {}
        f:spec:
          .: {}
          f:diskFileSpecMap:
            .: {}
            f:0e762d26-0c69-47f2-9c65-8ab5e1723f44:
              .: {}
              f:evictionRequested: {}
            f:48254039-3639-4916-8f77-068b9e663872:
              .: {}
              f:evictionRequested: {}
            f:c8e8efeb-6c05-4854-b230-015ad8888ecc:
              .: {}
              f:evictionRequested: {}
          f:disks: {}
          f:minNumberOfCopies: {}
          f:sourceParameters:
            .: {}
            f:url: {}
          f:sourceType: {}
      manager: support-bundle-kit-amd64
      operation: Update
      time: "2024-10-28T02:27:59Z"
    - apiVersion: longhorn.io/v1beta2
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          .: {}
          f:checksum: {}
          f:diskFileStatusMap:
            .: {}
            f:0e762d26-0c69-47f2-9c65-8ab5e1723f44:
              .: {}
              f:lastStateTransitionTime: {}
              f:progress: {}
              f:state: {}
            f:48254039-3639-4916-8f77-068b9e663872:
              .: {}
              f:lastStateTransitionTime: {}
              f:progress: {}
              f:state: {}
            f:c8e8efeb-6c05-4854-b230-015ad8888ecc:
              .: {}
              f:lastStateTransitionTime: {}
              f:progress: {}
              f:state: {}
          f:diskLastRefAtMap:
            .: {}
            f:0e762d26-0c69-47f2-9c65-8ab5e1723f44: {}
            f:48254039-3639-4916-8f77-068b9e663872: {}
            f:c8e8efeb-6c05-4854-b230-015ad8888ecc: {}
          f:ownerID: {}
          f:size: {}
          f:uuid: {}
          f:virtualSize: {}
      manager: support-bundle-kit-amd64
      operation: Update
      subresource: status
      time: "2024-10-28T02:27:59Z"
    name: vmi-aad51d2a-c8f3-4907-9fd0-64d77b29cbc7
    namespace: longhorn-system
    resourceVersion: "2441"
    uid: 4d617e27-756c-4e03-918a-b37c85b7fcf9
  spec:
    diskFileSpecMap:
      0e762d26-0c69-47f2-9c65-8ab5e1723f44:
        evictionRequested: false
      48254039-3639-4916-8f77-068b9e663872:
        evictionRequested: false
      c8e8efeb-6c05-4854-b230-015ad8888ecc:
        evictionRequested: false
    disks: {}
    minNumberOfCopies: 3
    sourceParameters:
      url: http://172.19.96.23/iso/sles/SLES15-SP4-Minimal-VM.x86_64-OpenStack-Cloud-GM.qcow2
    sourceType: download
  status:
    checksum: 37ae4631545dcbf12832636368abadcf7b346526aadfd1f4279c9c64ec33d95f491ad87dedc6656f519797bd79bf15dfcac0e74df33d4a69fd695c630d36774f
    diskFileStatusMap:
      0e762d26-0c69-47f2-9c65-8ab5e1723f44:
        lastStateTransitionTime: "2024-10-27T07:33:53Z"
        progress: 100
        state: ready
      48254039-3639-4916-8f77-068b9e663872:
        lastStateTransitionTime: "2024-10-27T07:48:29Z"
        progress: 100
        state: ready
      c8e8efeb-6c05-4854-b230-015ad8888ecc:
        lastStateTransitionTime: "2024-10-27T09:08:12Z"
        progress: 100
        state: ready
    diskLastRefAtMap:
      0e762d26-0c69-47f2-9c65-8ab5e1723f44: "2024-10-27T06:50:45Z"
      48254039-3639-4916-8f77-068b9e663872: "2024-10-27T06:50:18Z"
      c8e8efeb-6c05-4854-b230-015ad8888ecc: "2024-10-27T09:07:54Z"
    ownerID: harvester-node-1
    size: 288577024
    uuid: 0b3cdb34
    virtualSize: 999292928
  ```

I will try to reproduce to check if the bi `DiskFileStatusMap` restructure always happens, and the Harvester controller should improve to sync with the bi state, not blocked by the following condition:
- https://github.com/harvester/harvester/blob/19bc22db86ec0c9d796ec1d9116c3ddce908082a/pkg/controller/master/image/backing_image_controller.go#L30-L32
- https://github.com/harvester/harvester/blob/19bc22db86ec0c9d796ec1d9116c3ddce908082a/pkg/controller/master/image/backing_image_controller.go#L59-L61

Besides, this seems to only happen to the image not used by any volume before the upgrade.

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.4-eaf3009a-head` (24/11/05). Close this issue. 

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Check none used image is active after upgrade to v1.4-head $~~$</summary>

1. Before upgrade, on v1.3.2 Harvester, all image state is `Active` 
  ![image](https://github.com/user-attachments/assets/336cc6da-cc30-47f5-a07f-e0f0511950aa)

2. We only use `ubuntu 22.04` image to create virtual machine with 
  ![image](https://github.com/user-attachments/assets/51ebd84d-2401-487b-92fa-3a86d0fc7d8c)

3. After upgraded to v1.4-head
  ![image](https://github.com/user-attachments/assets/efaaf40d-3e0d-4fb5-af5b-dfef9f94d9bd)

4. Confirm all image state is `Active` after upgrade complete 
  ![image](https://github.com/user-attachments/assets/0827bc9d-46d2-42fc-ac2f-8aa202b4d0cc)

5. Can use all images to create vm 
  ![image](https://github.com/user-attachments/assets/b8a44848-4442-4747-902e-8f8ec544cc21)



 </details>


### Test Information
* Test Environment: Three nodes kvm machines
* Harvester version: `v1.4-eaf3009a-head` (24/11/05)

### Verify Steps
<details><summary>Harvester installation path</summary>

  1. Prepare three nodes v1.3.2 Harvester cluster 
  2. Create several image with different operating system and file types (qcow2, img and, iso)
  3. Only use one of the image to create a virtual machine
  4. Keep the other three images not used for any vm and volume 
  5. Upgrade to v1.4-head
  6. After upgrade complete, check the image state
  7. All images should be active and capable to use for vm and volume 

</details>
